Blejn/TwitterClone
# Feature/cleaning files

## Description

This Pull Request (PR) adds several enhancements to the registration form and user profile functionality. The main changes include:

1. **Error Handling in Registration Form**
   - Improved error handling for the registration form.
   - Clearer error messages and validation checks to ensure a smoother user experience during registration.

2. **Profile Picture Upload**
   - Added the ability for users to upload a profile picture during registration.
   - If a user has a profile picture, it will be displayed in the navigation bar after login.

## How to Test

To test these changes, follow these steps:

1. Navigate to the registration page.
2. Fill in the registration form, including the option to upload a profile picture.
3. Submit the form with valid and invalid data to observe the error handling.
4. After successful registration, log in to the application.
5. Observe the navigation bar; if you uploaded a profile picture during registration, it should be displayed there.

## Checklist

Please review and ensure that the following tasks have been completed:

- [x] Error handling in the registration form has been implemented.
- [x] Users can upload a profile picture during registration.
- [x] The profile picture is displayed in the navigation bar if available.
- [x] Unit tests have been added or updated to cover these changes.
- [x] Documentation (if any) has been updated to reflect these enhancements.


## Checklist

- [x] I have reviewed the code and provided constructive feedback.
- [x] I have tested the changes locally and they function as expected.
- [x] All checks and tests have passed.
